pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -369,31 +369,31 @@ int kprobe__udp_recvmsg(struct pt_regs* ctx) {
     struct sock* sk = (struct sock*)PT_REGS_PARM1(ctx);
     struct msghdr* msg = (struct msghdr*)PT_REGS_PARM2(ctx);
     int flags = (int)PT_REGS_PARM5(ctx);
-    return handle_udp_recvmsg(sk, msg, flags, &udp_recv_sock);
+    return handle_udp_recvmsg(sk, msg, flags, (struct bpf_map_def *)&udp_recv_sock);
```

Comment:
Is it possible to define a `struct bpf_map_def *` right after the map definition with the actual name of the map so we can avoid the cast here? i.e. the macro could produce something like:
```c
struct {...} _udp_recv_sock SEC(.maps);
struct bpf_map_def * udp_recv_sock = &_udp_recv_sock;
```

Reply:
We could also relax the argument type to `void *`. That is what the underlying map helpers use.

Reply:
Having bpf_map_def* improves readability imo.

Reply:
I'm not sure you can have both since they have different contents.